apps/extension/src/background/keyring/keyring.ts

```diff
@@ -133,8 +150,65 @@ export class KeyRing {
     return false;
   }
 
+  public static deriveTransparentAccount(
+    seed: Uint8Array,
+    path: Bip44Path,
+    parentId: string
+  ): DerivedAccountInfo {
+    const { account, change, index = 0 } = path;
+    const root = "m/44'";
+    const { coinType } = chains[0].bip44;
+    const derivationPath = [
+      root,
+      `${coinType}'`,
+      `${account}`,
+      change,
+      index,
+    ].join("/");
+    const bip44 = new HDWallet(seed);
+    const derivedAccount = bip44.derive(derivationPath);
+
+    const address = new Address(derivedAccount.private().to_hex()).implicit();
+    const id = getId("account", parentId, account, change, index);
+    const text = derivedAccount.private().to_hex();
+
+    return {
+      address,
+      id,
+      text,
+    };
+  }
+
+  public static deriveShieldedAccount(
+    seed: Uint8Array,
+    path: Bip44Path,
+    parentId: string
+  ): DerivedAccountInfo {
+    const { index = 0 } = path;
+    const id = getId("shielded-account", parentId, index);
+    const zip32 = new ShieldedHDWallet(seed);
+    const account = zip32.derive_to_serialized_keys(index);
```

Comment:
i'm curious if it is possible(and makes sense) to return those those `masp::` types from this function? Because at the moment(if I understand correctly :)) we are serializing `xsk` `xfvk` and `payment_address` and deserializing them few lines below in the constructor of `ExtendedSpendingKey` and other classes.

Reply:
I will take a look! They will need to be converted to a JSON serializable value before returning to TS (string or byte array), but I'll see if this can be cleaned up a bit.